src/Files.App/Helpers/GitHelpers.cs

```diff
@@ -266,6 +268,104 @@ public static async void PullOrigin(string? repositoryPath)
 			IsExecutingGitAction = false;
 		}
 
+		public static bool IsRepositoryEx(string path, out string repoRootPath)
+		{
+			repoRootPath = path;
+
+			var rootPath = SystemIO.Path.GetPathRoot(path);
+			if (rootPath is null)
+				return false;
+
+			var repositoryRootPath = GetGitRepositoryPath(path, rootPath);
+			if (string.IsNullOrEmpty(repositoryRootPath))
+				return false;
+
+			if (Repository.IsValid(repositoryRootPath))
+			{
+				repoRootPath = repositoryRootPath;
+				return true;
+			}
+
+			return false;
+		}
+
+		public static GitItemModel GetGitInformationForItem(Repository repository, string path)
+		{
+			var rootRepoPath = repository.Info.WorkingDirectory;
+			var relativePath = path.Substring(rootRepoPath.Length).Replace('\\', '/');
+
+			var commit = GetLastCommitForFile(repository, relativePath);
+			//var commit = repository.Commits.QueryBy(relativePath).FirstOrDefault()?.Commit; // Considers renames but slow
+
+			var changeKind = ChangeKind.Unmodified;
+			//foreach (TreeEntryChanges c in repository.Diff.Compare<TreeChanges>())
+			foreach (TreeEntryChanges c in repository.Diff.Compare<TreeChanges>(repository.Commits.FirstOrDefault()?.Tree, DiffTargets.Index | DiffTargets.WorkingDirectory))
+			{
+				if (c.Path.StartsWith(relativePath))
+				{
+					changeKind = c.Status;
+					break;
+				}
+			}
+
+			string? changeKindHumanized = "";
+			if (changeKind is not ChangeKind.Ignored)
+			{
+				changeKindHumanized = changeKind switch
+				{
+					ChangeKind.Added => "A",
+					ChangeKind.Deleted => "D",
+					ChangeKind.Modified => "M",
+					ChangeKind.Untracked => "U",
+					_ => "",
+				};
+			}
+
+			var gitItemModel = new GitItemModel()
+			{
+				Status = changeKind,
+				StatusHumanized = changeKindHumanized,
+				LastCommit = commit,
+				Path = relativePath,
+			};
+
+			return gitItemModel;
+		}
+
+		private static Commit? GetLastCommitForFile(Repository repository, string currentPath)
+		{
+			foreach (var currentCommit in repository.Commits)
+			{
+				var currentTreeEntry = currentCommit.Tree[currentPath];
+				if (currentTreeEntry == null)
+					return null;
+
+				var parentCount = currentCommit.Parents.Take(2).Count();
+				if (parentCount == 0)
+				{
+					return currentCommit;
+				}
+				else if (parentCount == 1)
+				{
+					var parentCommit = currentCommit.Parents.Single();
+
+					// Does not consider renames
+					var parentPath = currentPath; 
+
+					var parentTreeEntry = parentCommit.Tree[parentPath];
+
+					if (parentTreeEntry == null ||
+						parentTreeEntry.Target.Id != currentTreeEntry.Target.Id ||
+						parentPath != currentPath)
+					{
+						return currentCommit;
+					}
```

Comment:
What is `parentPath` needed for? You never change neither it nor `currentPath`

Reply:
i didnt created this method. let me check after 2.5

Reply:
@gave92 did you create this?

Reply:
Let me change it later